src/Database/Adapter/SQLite.php

```
@@ -1939,6 +1944,28 @@ public function getSupportForTTLIndexes(): bool
+        return $this->onDuplicate === OnDuplicate::Fail
+            ? 'INSERT INTO'
+            : 'INSERT OR IGNORE INTO';
+    }
+
+    /**
+     * SQLite realizes Upsert via the `INSERT OR REPLACE` keyword, so no SUFFIX
+     * clause is needed. Override MariaDB's `ON DUPLICATE KEY UPDATE` suffix.
+     *
+     * @param array<string> $columns
+     */
+    protected function getInsertSuffix(string $table, array $columns = []): string
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$table'. (undefined)

(UnusedFormalParameter)